### ISSUE TITLE
Error: Could not find Chromium (rev. 1108766)

### ISSUE THREAD

**Issue by ubiman** [mermaid-js/mermaid-cli#650]
**The script** (file: `input.mmd`):
```
flowchart LR
  A-->B
```

**The command line**:
```sh
mmdc -i input.mmd -o input.png
```

**The error message**:
```
Error: Could not find Chromium (rev. 1108766). This can occur if either
 1. you did not perform an installation before running the script (e.g. `npm install`) or
 2. your cache path is incorrectly configured (which is: /home/jfr/.cache/puppeteer).
For (2), check out our guide on configuring puppeteer at https://pptr.dev/guides/configuration.
    at ChromeLauncher.resolveExecutablePath (file:///usr/lib/node_modules/@mermaid-js/mermaid-cli/node_modules/puppeteer-core/lib/esm/puppeteer/node/ProductLauncher.js:263:27)
    at ChromeLauncher.executablePath (file:///usr/lib/node_modules/@mermaid-js/mermaid-cli/node_modules/puppeteer-core/lib/esm/puppeteer/node/ChromeLauncher.js:176:25)
    at ChromeLauncher.computeLaunchArguments (file:///usr/lib/node_modules/@mermaid-js/mermaid-cli/node_modules/puppeteer-core/lib/esm/puppeteer/node/ChromeLauncher.js:93:37)
    at async ChromeLauncher.launch (file:///usr/lib/node_modules/@mermaid-js/mermaid-cli/node_modules/puppeteer-core/lib/esm/puppeteer/node/ProductLauncher.js:57:28)
    at async run (file:///usr/lib/node_modules/@mermaid-js/mermaid-cli/src/index.js:404:19)
    at async cli (file:///usr/lib/node_modules/@mermaid-js/mermaid-cli/src/index.js:184:3)
```

**The environment**:
 * Recent computer (less than 6 months, 4Ghz 8Cores RAM:32GB SSD:1TB )
 * Ubuntu 22.04.3 (up to date, stable)
 * NPM version 10.4.0
 * NodeJS v21.6.1
 * Chromium 121.0.6167.160 snap
 * Mermaid-cli (`mmdc -V`) 10.8.0
 * `puppeteer --version` returns `unknown` (!?!)

**When installing mermaid-cli** (`@latest`):
```
 sudo npm install -g @mermaid-js/mermaid-cli@latest

 npm WARN deprecated puppeteer@19.11.1: < 21.5.0 is no longer supported
```

I did reinstall and reboot everything many times. I've looked everything and everywhere I could. I've spent many many hours. I am a little desperate. I don't know what else to do.

Please, any help will be very much appreciated. **Thank you**!!!

**Comment by tautschnig** [comment]
For all that I can tell the fix is to upgrade the puppeteer dependency. See #627 (and #646).

**Comment by ubiman** [comment]
Thank you for the fast reply. Appreciated!

I did replicate the change in "package.json" mentioned in #646. It **didn't** fix the error when running _mmdc_. I tried reinstalling mermaid-cli afterward just in case, but as I suspected, it just wiped out my manual update to package.json. Bottom line, I'm not skilled enough to understand what I am doing here.

So, I downgraded my way down to mermaid-cli 9.1.7. Although, the install still complains about the puppeteer version (the version on my machine is 22.0.0), **the _mmdc_ command now works** with mermaid 9.1.7. I am OK with this workaround.

Since I seem to be the only one having this issue, I probably messed up something in my first installation attempt. Apologies.

**Comment by aloisklink** [comment]
@ubiman, can you try running `npx @mermaid-js/mermaid-cli@latest`?

You can use `echo 'info' | npx @mermaid-js/mermaid-cli --input '-' --output 'out.svg'` to test rendering a super basic `info` diagram.

My gut feeling is that the issue is with `sudo`.

`npx` is quite nice, since it automatically installs `@mermaid-js/mermaid-cli` into a cache directory if it's not already installed.

---

> For all that I can tell the fix is to upgrade the puppeteer dependency. See https://github.com/mermaid-js/mermaid-cli/issues/627 (and https://github.com/mermaid-js/mermaid-cli/pull/646).

That's just a deprecation warning, it should work fine on the current version (the older version is even faster, due to some changes in the newer versions of puppeteer).

**Comment by ubiman** [comment]
@aloisklink, using _npx_ with the latest version, no output is produced. It doesn't show any error message though. Not sure why. Here's the full command and the full result of it:

```sh
$ npx @mermaid-js/mermaid-cli@latest -i input.mmd -o out.svg
Need to install the following packages:
@mermaid-js/mermaid-cli@10.8.0
Ok to proceed? (y) y
npm WARN deprecated puppeteer@19.11.1: < 21.5.0 is no longer supported
```

As with _mmdc_, if I execute _npx_ using version 9, then it works. Here's the command and result:

```sh
$ npx @mermaid-js/mermaid-cli@9.0.0 -i input.mmd -o out.svg
Need to install the following packages:
@mermaid-js/mermaid-cli@9.0.0
Ok to proceed? (y) y
npm WARN deprecated puppeteer@13.7.0: < 21.5.0 is no longer supported
Generating single mermaid chart
```

All my tests show that `mermaid-cli 9.1.7` and versions below work for me. Everything above `mermaid-cli 9.1.7` seems to be incompatible with _something_ in my environment.

I would love to use the newer mermaid version as some of my diagrams use the `~~~` connector (apparently not supported in 9.1.7), but the older version is better than nothing.

Thank you for your support!

**Comment by ubiman** [comment]
Ok. **All is working now**.

I tried deleting the puppeteer cache, but it didn't change the result I was getting.

Since I remembered installing the latest version of puppeteer and that it was higher than 19.11.1, I researched for a way to list the packages installed with NPM. That led me to do the following command:

```sh
$ npm list
usr@ /home/usr
├── @mermaid-js/mermaid-cli@10.8.0
├── mermaid@10.8.0
└── puppeteer@22.1.0
```

That is when I saw `mermaid@10.8.0` which I believe is something different than `mermaid-cli` and that I installed at the very beginning (pretty sure I installed it before mermaid-cli) and forgot about.

After removing  `mermaid` and reinstalling `mermaid-cli`, it started to work! Not a 100% sure that having `mermaid` beside `mermaid-cli` was the root cause of my problem, but I can't see anything else.

NOTE: the warning message `npm WARN deprecated puppeteer@19.11.1: < 21.5.0 is no longer supported` is still displayed when installing mermaid-cli, but it doesn't seem to cause an issue.

Many thanks to @aloisklink and @tautschnig for their help and generosity with their time!!!
